storage/clickhousespanstore/writer_test.go

```diff
@@ -0,0 +1,308 @@
+
+	spyLogger := mocks.NewSpyLogger()
+	spanWriter := getSpanWriter(spyLogger, db, EncodingJSON)
+
+	spans := generateRandomSpans()
+	expectIndexWritten(mock, spans, spanWriter)
+	if err = spanWriter.writeIndexBatch(spans); err != nil {
+		t.Fatalf("Could not write spans due to error: %s", err)
+	}
+	assertExpectationsWereMet(t, mock)
+	spyLogger.AssertLogsEmpty(t)
+}
+
+func expectModelWritten(
+	mock sqlmock.Sqlmock,
+	spans []*model.Span,
+	marshal func(span *model.Span) ([]byte, error),
+	spanWriter *SpanWriter,
+) error {
+	mock.ExpectBegin()
+	prep := mock.ExpectPrepare(fmt.Sprintf("INSERT INTO %s (timestamp, traceID, model) VALUES (?, ?, ?)", spanWriter.spansTable))
+	for _, span := range spans {
+		serializedSpan, err := marshal(span)
+		if err != nil {
+			return fmt.Errorf("could not marshal %s due to %s", fmt.Sprint(span), err)
+		}
+
+		prep.
+			ExpectExec().
+			WithArgs(
+				span.StartTime,
+				span.TraceID.String(),
+				serializedSpan,
+			).
+			WillReturnResult(sqlmock.NewResult(1, 1))
+	}
+	mock.ExpectCommit()
+	return nil
+}
+
+func expectIndexWritten(
+	mock sqlmock.Sqlmock,
+	spans []*model.Span,
+	spanWriter *SpanWriter,
+) {
+	mock.ExpectBegin()
+	prep := mock.ExpectPrepare(fmt.Sprintf(
+		"INSERT INTO %s (timestamp, traceID, service, operation, durationUs, tags) VALUES (?, ?, ?, ?, ?, ?)",
+		spanWriter.indexTable,
+	))
+	for _, span := range spans {
+		prep.
+			ExpectExec().
+			WithArgs(
+				span.StartTime,
+				span.TraceID,
+				span.Process.ServiceName,
+				span.OperationName,
+				span.Duration.Microseconds(),
+				fmt.Sprint(uniqueTagsForSpan(span)),
+			).
+			WillReturnResult(sqlmock.NewResult(1, 1))
+	}
+	mock.ExpectCommit()
+}
+
+func getDbMock() (*sql.DB, sqlmock.Sqlmock, error) {
+	return sqlmock.New(
+		sqlmock.QueryMatcherOption(sqlmock.QueryMatcherEqual),
+		sqlmock.ValueConverterOption(mocks.ConverterMock{}),
+	)
+}
+
+func getSpanWriter(spyLogger mocks.SpyLogger, db *sql.DB, encoding Encoding) *SpanWriter {
+	return NewSpanWriter(
+		spyLogger,
+		db,
+		testIndexTable,
+		testSpansTable,
+		encoding,
+		0,
+		0,
+	)
+}
+
+func generateRandomSpans() []*model.Span {
+	spans := make([]*model.Span, testSpanCount)
+	for i := 0; i < testSpanCount; i++ {
+		span := generateRandomSpan()
+		spans[i] = &span
+	}
+	return spans
+}
+
+func generateRandomSpan() model.Span {
+	processTags := generateRandomTags()
+	process := model.Process{
+		ServiceName: "service" + strconv.FormatUint(rand.Uint64(), 10),
+		Tags:        processTags,
+	}
+	span := model.Span{
+		TraceID:       model.NewTraceID(rand.Uint64(), rand.Uint64()),
+		SpanID:        model.NewSpanID(rand.Uint64()),
+		OperationName: "operation" + strconv.FormatUint(rand.Uint64(), 10),
+		StartTime:     time.Unix(rand.Int63n(time.Now().Unix()), 0),
+		Process:       &process,
+		Tags:          generateRandomTags(),
+		Duration:      time.Unix(rand.Int63n(1<<32), 0).Sub(time.Unix(0, 0)),
+	}
+	return span
+}
+
+func generateRandomTags() []model.KeyValue {
```

Comment:
the function is not generating any random tags. 

I would suggest adding a parameter `count` to the signature.

Reply:
> the function is not generating any random tags.

It generates pseudo-random tags, I think it's useful for re-testing.

> I would suggest adding a parameter `count` to the signature.

I'm passing only constant to this parameter, so linter falis and suggests to unparam this.